---
title: "Lung Metagenomic Analysis"
author: "Rebecca Weiss"
date: "3/30/2021"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## To load the metagenomeSeq library:
```{r}
library(metagenomeSeq)
```



## Reading in a biom file
```{r message=FALSE, warning=FALSE}
library(biomformat)
biom_file <- system.file("extdata", "min_sparse_otu_table.biom",
package = "biomformat")
b <- read_biom(biom_file)
biom2MRexperiment(b)
```

## Loading the count data
```{r}
dataDirectory <- system.file("extdata", package = "metagenomeSeq")
lung = loadMeta(file.path(dataDirectory, "CHK_NAME.otus.count.csv"))
dim(lung$counts)
```

## Loading the taxonomy
```{r}
taxa = read.delim(file.path(dataDirectory, "CHK_otus.taxonomy.csv"),
stringsAsFactors = FALSE)
```

## Loading the metadata
```{r message=FALSE, warning=FALSE}
clin = loadPhenoData(file.path(dataDirectory, "CHK_clinical.csv"),
tran = TRUE)
ord = match(colnames(lung$counts), rownames(clin))
clin = clin[ord, ]
head(clin[1:2, ])
```

## Creating an MRExperiment object
```{r message=FALSE, warning=FALSE}
phenotypeData = AnnotatedDataFrame(clin)
phenotypeData

OTUdata = AnnotatedDataFrame(taxa)
OTUdata

obj = newMRexperiment(lung$counts,phenoData=phenotypeData,featureData=OTUdata)
# Links to a paper providing further details can be included optionally.
# experimentData(obj) = annotate::pmid2MIAME("21680950")
obj
```

## Example datasets
```{r message=FALSE, warning=FALSE}
data(lungData)
lungData

data(mouseData)
mouseData
```

# Useful commands 
## Some examples:
```{r message=FALSE, warning=FALSE}
phenoData(obj)

head(pData(obj), 3)

featureData(obj)
```


## Subsetting an MRExperiment class
```{r message=FALSE, warning=FALSE}
featuresToKeep = which(rowSums(obj) >= 100)
samplesToKeep = which(pData(obj)$SmokingStatus == "Smoker")
obj_smokers = obj[featuresToKeep, samplesToKeep]
obj_smokers
```

## Normalization factors
```{r message=FALSE, warning=FALSE}
head(normFactors(obj))
normFactors(obj) <- rnorm(ncol(obj))
head(normFactors(obj))
```

## Library sizes (sequencing depths) can be accessed or replaced with the libSize method
```{r message=FALSE, warning=FALSE}
head(libSize(obj))
libSize(obj) <- rnorm(ncol(obj))
head(libSize(obj))
```

## Data can be filtered to maintain a threshold of minimum depth or OTU presence
```{r message=FALSE, warning=FALSE}
data(mouseData)
filterData(mouseData, present = 10, depth = 1000)
```

## Two MRexperiment-class objects can be merged with the mergeMRexperiments function, e.g.:
```{r message=FALSE, warning=FALSE}
data(mouseData)
newobj = mergeMRexperiments(mouseData, mouseData)
```

# Normalization 
## Calculcation normalization factors
```{r message=FALSE, warning=FALSE}
data(lungData)
p = cumNormStatFast(lungData)

lungData = cumNorm(lungData, p = p)
```

## Calculating using wrench
```{r message=FALSE, warning=FALSE}
condition = mouseData$diet
mouseData = wrenchNorm(mouseData, condition = condition)
```

## Exporting the data
```{r message=FALSE, warning=FALSE}
mat = MRcounts(lungData, norm = TRUE, log = TRUE)[1:5, 1:5]
exportMat(mat, file = file.path(dataDirectory, "tmp.tsv"))

exportStats(lungData[, 1:5], file = file.path(dataDirectory,
"tmp.tsv"))
## Default value being used.
head(read.csv(file = file.path(dataDirectory, "tmp.tsv"), sep = "\t"))
```

# Statistical Testing
## Example using fitFeatureModel for differential abundance testing
```{r message=FALSE, warning=FALSE}
data(lungData)
lungData = lungData[, -which(is.na(pData(lungData)$SmokingStatus))]
lungData = filterData(lungData, present = 30, depth = 1)
lungData <- cumNorm(lungData, p = 0.5)
pd <- pData(lungData)
mod <- model.matrix(~1 + SmokingStatus, data = pd)
lungres1 = fitFeatureModel(lungData, mod)
head(MRcoefs(lungres1))
```


## Example using fitZig for differential abundance testing
```{r message=FALSE, warning=FALSE}
data(lungData)
controls = grep("Extraction.Control", pData(lungData)$SampleType)
lungTrim = lungData[, -controls]
rareFeatures = which(rowSums(MRcounts(lungTrim) > 0) < 10)
lungTrim = lungTrim[-rareFeatures, ]
lungp = cumNormStat(lungTrim, pFlag = TRUE, main = "Trimmed lung data")

lungTrim = cumNorm(lungTrim, p = lungp)

smokingStatus = pData(lungTrim)$SmokingStatus
bodySite = pData(lungTrim)$SampleType
normFactor = normFactors(lungTrim)
normFactor = log2(normFactor/median(normFactor) + 1)
mod = model.matrix(~smokingStatus + bodySite + normFactor)
settings = zigControl(maxit = 10, verbose = TRUE)
fit = fitZig(obj = lungTrim, mod = mod, useCSSoffset = FALSE,
control = settings)
```


## Multiple groups
```{r message=FALSE, warning=FALSE}
# maxit=1 is for demonstration purposes
settings = zigControl(maxit = 1, verbose = FALSE)
mod = model.matrix(~bodySite)
colnames(mod) = levels(bodySite)
# fitting the ZIG model
res = fitZig(obj = lungTrim, mod = mod, control = settings)
# The output of fitZig contains a list of various useful
# items. hint: names(res). Probably the most useful is the
# limma 'MLArrayLM' object called fit.
zigFit = slot(res, "fit")
finalMod = slot(res, "fit")$design
contrast.matrix = makeContrasts(BAL.A - BAL.B, OW - PSB, levels = finalMod)
fit2 = contrasts.fit(zigFit, contrast.matrix)
fit2 = eBayes(fit2)
topTable(fit2)
```

## Exporting fits
```{r message=FALSE, warning=FALSE}
taxa = sapply(strsplit(as.character(fData(lungTrim)$taxa), split = ";"),
function(i) {
i[length(i)]
})
head(MRcoefs(fit, taxa = taxa, coef = 2))
```


##  Log Normal permutation test
```{r message=FALSE, warning=FALSE}
coeffOfInterest = 2
res = fitLogNormal(obj = lungTrim, mod = mod, useCSSoffset = FALSE,
B = 10, coef = coeffOfInterest)
# extract p.values and adjust for multiple testing res$p are
# the p-values calculated through permutation
adjustedPvalues = p.adjust(res$p, method = "fdr")
# extract the absolute fold-change estimates
foldChange = abs(res$fit$coef[, coeffOfInterest])
# determine features still significant and order by the
sigList = which(adjustedPvalues <= 0.05)
sigList = sigList[order(foldChange[sigList])]
# view the top taxa associated with the coefficient of
# interest.
head(taxa[sigList])
```

## Presence-absence testing
```{r message=FALSE, warning=FALSE}
classes = pData(mouseData)$diet
res = fitPA(mouseData[1:5, ], cl = classes)
# Warning - the p-value is calculating 1 despite a high odd's
# ratio.
head(res)
```

## Discovery odds ratio testing
```{r message=FALSE, warning=FALSE}
classes = pData(mouseData)$diet
res = fitDO(mouseData[1:100, ], cl = classes, norm = FALSE, log = FALSE)
head(res)
```

## Feature correlations
```{r message=FALSE, warning=FALSE}
cors = correlationTest(mouseData[55:60, ], norm = FALSE, log = FALSE)
head(cors)
```

## Unique OTUs or features
```{r message=FALSE, warning=FALSE}
cl = pData(mouseData)[["diet"]]
uniqueFeatures(mouseData, cl, nsamples = 10, nreads = 100)
```

# Aggregating counts
```{r message=FALSE, warning=FALSE}
obj = aggTax(mouseData, lvl = "phylum", out = "matrix")
head(obj[1:5, 1:5])
```


```{r message=FALSE, warning=FALSE}
obj = aggSamp(mouseData, fct = "mouseID", out = "matrix")
head(obj[1:5, 1:5])
```

# Visualization of features
```{r message=FALSE, warning=FALSE}
trials = pData(mouseData)$diet
heatmapColColors = brewer.pal(12, "Set3")[as.integer(factor(trials))]
heatmapCols = colorRampPalette(brewer.pal(9, "RdBu"))(50)
```

## Plot MRHeatmap
```{r message=FALSE, warning=FALSE}
plotMRheatmap(obj = mouseData, n = 200, cexRow = 0.4, cexCol = 0.4,
trace = "none", col = heatmapCols, ColSideColors = heatmapColColors)
# plotCorr
plotCorr(obj = mouseData, n = 200, cexRow = 0.25, cexCol = 0.25,
trace = "none", dendrogram = "none", col = heatmapCols)
```

## Example of plotting CMDS plots of the data and the rarefaction effect at the OTU level
```{r message=FALSE, warning=FALSE}
cl = factor(pData(mouseData)$diet)
# plotOrd - can load vegan and set distfun = vegdist and use
# dist.method='bray'
plotOrd(mouseData, tran = TRUE, usePCA = FALSE, useDist = TRUE,
bg = cl, pch = 21)
# plotRare
res = plotRare(mouseData, cl = cl, pch = 21, bg = cl)
# Linear fits for plotRare / legend
tmp = lapply(levels(cl), function(lv) lm(res[, "ident"] ~ res[,
"libSize"] - 1, subset = cl == lv))
for (i in 1:length(levels(cl))) {
abline(tmp[[i]], col = i)
}
legend("topleft", c("Diet 1", "Diet 2"), text.col = c(1, 2),
box.col = NA)
```

## Feature specific
```{r message=FALSE, warning=FALSE}
head(MRtable(fit, coef = 2, taxa = 1:length(fData(lungTrim)$taxa)))
```


```{r message=FALSE, warning=FALSE}
patients = sapply(strsplit(rownames(pData(lungTrim)), split = "_"),
function(i) {
i[3]
})
pData(lungTrim)$patients = patients
classIndex = list(smoker = which(pData(lungTrim)$SmokingStatus ==
"Smoker"))
classIndex$nonsmoker = which(pData(lungTrim)$SmokingStatus ==
"NonSmoker")
otu = 779
# plotOTU
plotOTU(lungTrim, otu = otu, classIndex, main = "Neisseria meningitidis")
# Now multiple OTUs annotated similarly
x = fData(lungTrim)$taxa[otu]
otulist = grep(x, fData(lungTrim)$taxa)
# plotGenus
plotGenus(lungTrim, otulist, classIndex, labs = FALSE, main = "Neisseria meningitidis")
lablist <- c("S", "NS")
axis(1, at = seq(1, 6, by = 1), labels = rep(lablist, times = 3))
```


```{r message=FALSE, warning=FALSE}
classIndex = list(Western = which(pData(mouseData)$diet == "Western"))
classIndex$BK = which(pData(mouseData)$diet == "BK")
otuIndex = 8770
# par(mfrow=c(1,2))
dates = pData(mouseData)$date
plotFeature(mouseData, norm = FALSE, log = FALSE, otuIndex, classIndex,
col = dates, sortby = dates, ylab = "Raw reads")
```

## Summary 
# Citing metagenomeSeq
```{r message=FALSE, warning=FALSE}
citation("metagenomeSeq")
```

# Session info
```{r message=FALSE, warning=FALSE}
sessionInfo()
```
